### ISSUE TITLE
ability to set any user attribute

### ISSUE THREAD

**Issue by zaro0508** [Sage-Bionetworks/jccli#74]
Currently we can only use the user set command to update a small set of user attributes: username, email, first name and last name.   I would like to be able to update all user attributes.

**Comment by cascadianblue** [comment]
Definitely. I assume create-user should have all these attributes as well. Do you have any ideas besides adding them all basically manually as `@click.option`s?

**Comment by cascadianblue** [comment]
I'm gonna leave this link to the [create-user API url](https://docs.jumpcloud.com/1.0/systemusers/create-a-system-user) here for future reference

**Comment by cascadianblue** [comment]
We probably want to be able to set these as filters on `user list` as well, I assume

**Comment by zaro0508** [comment]
create with multiple user attributes and filtering is not a requirement.  I really just wan to set any and all user attributes.

To implement I would look at [click mulitple options](https://click.palletsprojects.com/en/7.x/options/#multiple-options).

Maybe pass in as key value pairs so something like this..
```
jccli set -attribute firstname=joe -a lastname=smith -a email=jsmith@acme.org
```

**Comment by cascadianblue** [comment]
So are you proposing that some of these fields (such as first name, last name, email, username) can be set as `--field value`s but the rest of them can only be done through `--attribute field=value`s?

Can the fields that can be set as `--field value` also be set by `--attribute field=value`? E.g. what happens when our client calls `jccli user set --username fakeguy123 --email new.email1@email.com --attribute email=new.email2@email.com`. What does `fakeguy123`'s email get set to?

**Comment by zaro0508** [comment]
How about we make required parameters (i.e. `--username`) fields and the rest need to be attributes?  So for your example a valid command would be `jccli user set --username fakeguy123 --attribute email=new.email2@email.com`.  if `--email` is passed in then it's an error.
